lib/OpenQA/WebAPI/Controller/API/V1/Iso.pm

```diff
@@ -327,39 +328,51 @@ defined. Internal method used by the B<schedule_iso()> method.
 sub job_create_dependencies {
     my ($self, $job, $testsuite_mapping, $created_jobs) = @_;
 
-    my @error_messages;
+    my (@error_messages, @info_messages);
     my $settings = $job->settings_hash;
     for my $dependency (
         ['START_AFTER_TEST', OpenQA::Schema::Result::JobDependencies::CHAINED],
         ['PARALLEL_WITH',    OpenQA::Schema::Result::JobDependencies::PARALLEL])
     {
         my ($depname, $deptype) = @$dependency;
         next unless defined $settings->{$depname};
-        for my $testsuite (_parse_dep_variable($settings->{$depname}, $settings)) {
-            if (!defined $testsuite_mapping->{$testsuite}) {
-                my $error_msg = "$depname=$testsuite not found - check for typos and dependency cycles";
+        for my $test (_parse_dep_variable($settings->{$depname}, $settings, testonly => 1)) {
+            if (!defined $testsuite_mapping->{$test}) {
+                my $error_msg = "$depname=$test not found - check for typos and dependency cycles";
                 OpenQA::Utils::log_warning($error_msg);
                 push(@error_messages, $error_msg);
             }
             else {
-                for my $parent (@{$testsuite_mapping->{$testsuite}}) {
-                    try {
-                        _check_for_cycle($job->id, $parent, $created_jobs);
+                my @machine = ($settings->{MACHINE});
+                if (!defined $testsuite_mapping->{$test}->{$machine[0]}) {
+                    my $info_msg = "$depname=$test:$machine[0] not found, but";
+                    @machine = keys %{$testsuite_mapping->{$test}};
+                    for (@machine) {
+                        $info_msg .= " $depname=$test:$_";
+                    }
+                    OpenQA::Utils::log_info($info_msg);
+                    push(@info_messages, $info_msg);
+                }
+                for my $machine (@machine) {
+                    for my $parent (@{$testsuite_mapping->{$test}->{$machine}}) {
+                        try {
+                            _check_for_cycle($job->id, $parent, $created_jobs);
```

Comment:
this goes a little deep here. This is (if I count correctly) for dependency, for test, for machine, for parent. Some of it should rather be moved into a function.